### PR TITLE
[UIPO-30] - Most popular component

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/MostPopular/MostPopular.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import ProductItem from './ProductItem';
+
+class MostPopular extends Component {
+  state = {
+    products: [],
+  }
+
+  async downloadProducts() {
+    const userApi = new HttpService();
+    const response = await userApi.get('http://localhost:4000/most-popular');
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js]: move url to config file

File: src/components/MostPopular/MostPopular.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    this.setState({ products: response.data });
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js]: what happens when you get an error in the response?

File: src/components/MostPopular/MostPopular.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import HttpService from '../../service/HttpService/httpService';
+import ProductItem from './ProductItem';
+
+class MostPopular extends Component {
+  state = {
+    products: [],
+  }
+
+  async downloadProducts() {
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js]: this method should be after componentDidMount

File: src/components/MostPopular/MostPopular.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    this.setState({ products: response.data });
+  }
+
+  componentDidMount() {
+    this.downloadProducts();
+  }
+
+  render() {
+    const { products } = this.state;
+    const block = products.map((product) => (
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js]: this should be a class method, for example `renderProducts`

File: src/components/MostPopular/MostPopular.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      <ProductItem key={product.id} product={product} />
+    ));
+    return (
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js:38]: add empty space before return

File: src/components/MostPopular/MostPopular.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      <div className="wrapper">
+        <h1 className="title">Most Popular</h1>
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js]: use bem

File: src/components/MostPopular/ProductItem/ProductItem.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import './ProductItem.scss';
+import cx from 'classnames';
+
+
+const ProductItem = ({ product }) => {
+  const { image, label, title, price } = product;
+
+  return (
+    <div className="product col-3">
+      <div className="product__img-wrapper">
```
Comment (Nazar-bnm) [inline src/components/MostPopular/ProductItem/ProductItem.js]: please assign `CN` const and use it like `className={${CN}__img-wrapper}`

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

File: src/components/MostPopular/MostPopular.js (hunk 2 above)
Comment (ghost) [reply src/components/MostPopular/MostPopular.js]: the component won't rendered

Comment (Nazar-bnm) [reply src/components/MostPopular/MostPopular.js]: please add some verify for response -> if (response && response.data) {... }

File: src/components/MostPopular/MostPopular.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+class MostPopular extends Component {
+  state = {
+    products: [],
+  }
+
+  componentDidMount() {
+    this.downloadProducts();
+  }
+
+  async downloadProducts() {
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js]: I think `getProducts` is better name for this method

File: src/components/MostPopular/MostPopular.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    const userApi = new HttpService();
+    const response = await userApi.get(urlBase);
+    this.setState({ products: response.data });
+  }
+
+  renderProducts() {
+    const { products } = this.state;
+    return products.map((product) => (
```
Comment (Nazar-bnm) [inline src/components/MostPopular/MostPopular.js:32]: add empty line before return

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

File: src/components/MostPopular/ProductItem/ProductItem.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+import './ProductItem.scss';
+import cx from 'classnames';
+// import { Transition } from 'react-transition-group';
```
Comment (Hollister009) [inline src/components/MostPopular/ProductItem/ProductItem.js]: remove this comment

File: src/components/MostPopular/ProductItem/ProductItem.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        {!isHovered && <div className={`${CN}__img-wrapper__labels`}>{label}</div>}
+        <img className={cx(`${CN}__img-wrapper__img`,
+          { [`${CN}__img-wrapper__img--hovered`]: isHovered })} src={image}/>
+      </div>
+      <div className={cx(`${CN}__title-wrapper`,
+        { [`${CN}__title-wrapper__title`]: isHovered })}>
+        <h2 className={cx(`${CN}__title-wrapper__title`,
+          { [`${CN}__title-wrapper__title--hovered`]: isHovered })}>{title}</h2>
+        {!isHovered && <h3 className={`${CN}__title-wrapper__price`}>{price}</h3>}
+        {isHovered && <span className='sizes'>Sizes: S - M - L </span>}
```
Comment (Hollister009) [inline src/components/MostPopular/ProductItem/ProductItem.js]: Shouldn't be any hardcoded values

File: src/components/MostPopular/ProductItem/ProductItem.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+import './ProductItem.scss';
+import cx from 'classnames';
+// import { Transition } from 'react-transition-group';
+
+const CN = 'product';
+const ProductItem = ({ product }) => {
+  const [isHovered, setHovered] = useState(false);
```
Comment (Hollister009) [inline src/components/MostPopular/ProductItem/ProductItem.js:9]: Why did you switch to hooks?

File: src/components/MostPopular/ProductItem/ProductItem.js (hunk 11 above)
Comment (Hollister009) [inline src/components/MostPopular/ProductItem/ProductItem.js]: Where did you get this code from?

File: src/components/MostPopular/ProductItem/ProductItem.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      {renderView(isHovered)}
+    </div>
+  );
+};
+
+ProductItem.propTypes = {
+  product: PropTypes.object,
```
Comment (Hollister009) [inline src/components/MostPopular/ProductItem/ProductItem.js]: Not very descriptive. What kind of object(shape) is product?

File: src/components/MostPopular/ProductItem/ProductItem.scss [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+      height: 306px;
+      padding: 50px;
+    }
+
+    &__flag {
+      width: 0;
+      height: 0;
+      border-style: solid;
+      border-width: 60px 60px 0 0;
+      border-color: transparent transparent transparent transparent;
```
Comment (Hollister009) [inline src/components/MostPopular/ProductItem/ProductItem.scss]: transparent x4?

**Review by Hollister009 — CHANGES_REQUESTED** [review]
You need to work on ProductItem component. Read my comments

File: src/components/MostPopular/ProductItem/ProductItem.js (hunk 12 above)
Comment (ghost) [reply src/components/MostPopular/ProductItem/ProductItem.js:9]: Taras advised me

File: src/components/MostPopular/ProductItem/ProductItem.js (hunk 11 above)
Comment (ghost) [reply src/components/MostPopular/ProductItem/ProductItem.js]: It was made by me)
I merged 2 method into 1 we have

File: src/components/MostPopular/ProductItem/ProductItem.js [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+  image: PropTypes.string,
+  label: PropTypes.string,
+  title: PropTypes.string,
+  price: PropTypes.string,
+  sizes: PropTypes.string,
```
Comment (Nazar-bnm) [inline src/components/MostPopular/ProductItem/ProductItem.js]:
```
product: PropTypes.shape({
    image: PropTypes.string,
    label: PropTypes.string,
    title: PropTypes.string,
    price: PropTypes.string,
    sizes: PropTypes.string
  }),
```

File: src/components/MostPopular/ProductItem/ProductItem.js (hunk 13 above)
Comment (Nazar-bnm) [reply src/components/MostPopular/ProductItem/ProductItem.js]: look example below

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

**Review by Nazar-bnm — APPROVED** [review]

**Pull request merged**